services/api/service.go

```diff
@@ -1335,9 +1333,47 @@ func (api *RelayAPI) handleSubmitNewBlock(w http.ResponseWriter, req *http.Reque
 		}
 	}()
 
+	// Get the current (in-memory) top bid value
+	api.builderTopBidValueLock.RLock()
```

Comment:
I think it would be useful to encapsulate this in a function, e.g
```
func (api *RelayAPI) topBidValue(parentHash string) (value *big.Int, ok bool) {
	api.builderTopBidValueLock.RLock()
	defer api.builderTopBidValueLock.RUnlock()
	value, ok = api.builderTopBidValue[parentHash]
	return
}
```

Reply:
on the fence about it. 3 lines and no repetition doesn't seem like it's worth it. if it's repeated 2 to 3 times then for sure.